Change over use of docker binary in Docker to use docker-py

a10y opened the pull request:
PR for #291 

@percyliang

Commit:
Change over use of docker binary in Docker to use docker-py

a10y commented:
Note: I've only tested this on docker 1.9.1, and seems to be working fine. I'm not certain if the features of the Docker Remote API are different enough across the recent versions of docker that it would lock us into using the most recent.

Commit:
Removed outdate comment, added useful comment

percyliang commented on requirements.txt:
```diff
@@ -1,7 +1,16 @@
 SQLAlchemy==1.0.8
-alembic==0.8.2
 watchdog==0.8.3
-# Use sckoo@stanford.edu's patched argcomplete, until pull request has been accepted
```
Keep comment

percyliang commented:
One tangential request: if user types `--request-docker-image <tab>`, then we should complete with things that begin with `codalab/...`, since novice users will want sane defaults here.  But if the user starts typing something that doesn't start with codalab, we can still do the completion as is done now.

Commit:
Trimmed down the requirements.txt, added psutil explicitly

I changed the requirements file back to what it was before, now adding
only docker-py and psutil so that we only pin top-level dependencies.
psutil was actually already being used in codalab_manager.py but was
not in the requirements.txt.

Commit:
When user enters nothing, complete for "codalab" by default

Commit:
Fixed outdated comment

a10y commented:
Last commit in, this PR should be GTG

percyliang commented on codalab/lib/completers.py:
```diff
@@ -183,8 +183,10 @@ class DockerImagesCompleter(CodaLabCompleter):
-        def handle_err(errcode, stderr):
-            warn("Error(%d): %s" % (errcode, stderr))
+        def handle_err(exc):
+            warn("Error: %s" % exc)
```
nit: use single quotes

percyliang commented:
LGTM!
